src/script.ts

```diff
@@ -77,6 +77,36 @@ class Script {
         throw err;
       }
     }
+  }
+
+  static fromBuffer(buf: Buffer, options: ScriptInitOptions) {
+    const br = new BufferReader(buf);
+    return this.fromBufferReader(br, options);
+  }
+
+  static fromBufferReader(
+    br: BufferReader,
+    options: ScriptInitOptions = { opreturn: false }
+  ) {
+    const chunks: ScriptChunk[] = [];
+
+    // TODO: Is it ok that these are undefined instead of false?
+    // This change needs reviewed
+    if (br.eof()) return options.opreturn ? undefined : new Script(br, chunks);
+    if (options.opreturn) {
+      let opcodenum = br.readUInt8();
+      if (opcodenum === Opcode.OP_FALSE) {
+        chunks.push({ opcodenum });
+        if (!br.eof()) {
+          opcodenum = br.readUInt8();
+        }
+      }
+      if (opcodenum !== Opcode.OP_RETURN) {
+        return undefined;
+      }
+      chunks.push({ opcodenum });
+    }
+    const script = new Script(br, chunks);
     return script;
```

Comment:
@kevinejohn see my TODO comment here. Before it was returning `false` instead of `undefined`. Would prefer to have it return `undefined` instead of `false` if possible.

Reply:
I forget exactly why I did it that way. Looking at the ways I use that method though we could just have it throw an error.